lib/embedded_ansible.rb

```diff
@@ -64,19 +64,20 @@ def self.services
     AwesomeSpawn.run!("source /etc/sysconfig/ansible-tower; echo $TOWER_SERVICES").output.split
   end
 
-  def self.playbook_extra_variables
-    json_value = {
+  def self.run_setup_script(exclude_tags)
+    json_extra_vars = {
       :minimum_var_space => 0,
       :http_port         => HTTP_PORT,
       :https_port        => HTTPS_PORT
     }.to_json
-    {:e => json_value}
-  end
-  private_class_method :playbook_extra_variables
 
-  def self.run_setup_script(params)
     with_inventory_file do |inventory_file_path|
-      AwesomeSpawn.run!(SETUP_SCRIPT, :params => params.merge(:i => inventory_file_path))
+      params = []
+      params << [:e, json_extra_vars]
+      params << [:i, inventory_file_path]
+      params << "--"
+      params << ["--skip-tags=", exclude_tags]
```

Comment:
Interesting, so the setup script doesn't take long options, but the setup script does. So we could do:

```ruby
params = {
  "--"         => nil,
  :extra_vars= => json_extra_vars,
  :inventory=  => inventory_file_path,
  :skip_tags=  => exclude_tags
}
```